Solutions/STM32F4DISCOVERY/scatterfile_gcc_missing_symbols.xml

```diff
@@ -1,5 +1,10 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <ScatterFile xmlns="http://schemas.microsoft.com/netmf/ScatterfileSchema.xsd">
+  <!-- 
+    - using newlib-nano adds libnosys.a, which calls sbrk, which requires the 'end' symbol
+    - this is pulled in by a _printf imported in function _Crypto_StepRSAOperation @ Crypto.lib(rsa_zenith.obj) 
+  -->
+  <Provide Name="end" />

```

Comment:
A little while ago I had a look at mBedTLS and it seems to offer up almost everything we need on the device side. On the desktop side the standard .NET framework libraries have matching counterparts. Unfortunately there's one snag. While I've been able to get a mBedTLS and .NET Framework to correctly match sign and verify operations with the RSA keys the hash algorithm the cryptolibs use in creating a signature doesn't appear to be a standard one. I've written some test utils to use the cryptolibs to compute the hash used with the RSA keys for creating the signature and have not found any standard hashing algorithm that matches the output of the cryptolibs. 

Thus, while it is possible to completely replace them (and OpenSSL) with mBedTLS on the device and normal .NET framework on the desktop side, there's currently no way to do so in a backwards compatible manner. Thus we'd need to drop Back Compat support or drag around the old desktop builds of the cryptolibs for use by MFDeploy when dealing with older devices.